packages/tempus/test/BDF2/Tempus_BDF2Test.cpp

```diff
@@ -191,6 +195,251 @@ TEUCHOS_UNIT_TEST(BDF2, SinCos)
+    // Read params from .xml file
+    RCP<ParameterList> pList =
+      getParametersFromXmlFile("Tempus_BDF2_SinCos.xml");
+
+    // Setup the SinCosModel
+    RCP<ParameterList> scm_pl = sublist(pList, "SinCosModel", true);
+    scm_pl->set("Use DfDp as Tangent", use_dfdp_as_tangent);
+    RCP<SinCosModel<double> > model =
+      Teuchos::rcp(new SinCosModel<double>(scm_pl));
+
+    dt /= 2;
+
+    // Setup sensitivities
+    RCP<ParameterList> pl = sublist(pList, "Tempus", true);
+    ParameterList& sens_pl = pl->sublist("Sensitivities");
+    if (use_combined_method)
+      sens_pl.set("Sensitivity Method", "Combined");
+    else {
+      sens_pl.set("Sensitivity Method", "Staggered");
+      sens_pl.set("Reuse State Linear Solver", true);
+    }
+    sens_pl.set("Use DfDp as Tangent", use_dfdp_as_tangent);
+
+    // Setup the Integrator and reset initial time step
+    pl->sublist("Default Integrator")
+       .sublist("Time Step Control").set("Initial Time Step", dt);
+    RCP<Tempus::IntegratorForwardSensitivity<double> > integrator =
+      Tempus::integratorForwardSensitivity<double>(pl, model);
+    order = integrator->getStepper()->getOrder();
+
+    // Initial Conditions
+    double t0 = pl->sublist("Default Integrator")
+      .sublist("Time Step Control").get<double>("Initial Time");
+    RCP<const Thyra::VectorBase<double> > x0 =
+      model->getExactSolution(t0).get_x();
+    const int num_param = model->get_p_space(0)->dim();
+    RCP<Thyra::MultiVectorBase<double> > DxDp0 =
+      Thyra::createMembers(model->get_x_space(), num_param);
+    for (int i=0; i<num_param; ++i)
+      Thyra::assign(DxDp0->col(i).ptr(),
+                    *(model->getExactSensSolution(i, t0).get_x()));
+    integrator->setInitialState(t0, x0, Teuchos::null, Teuchos::null,
+                                DxDp0, Teuchos::null, Teuchos::null);
+
+    // Integrate to timeMax
+    bool integratorStatus = integrator->advanceTime();
+    TEST_ASSERT(integratorStatus)
+
+    // Test if at 'Final Time'
+    double time = integrator->getTime();
+    double timeFinal =pl->sublist("Default Integrator")
+       .sublist("Time Step Control").get<double>("Final Time");
+    TEST_FLOATING_EQUALITY(time, timeFinal, 1.0e-14);
+
+    // Time-integrated solution and the exact solution
+    RCP<const Thyra::VectorBase<double> > x = integrator->getX();
+    RCP<const Thyra::MultiVectorBase<double> > DxDp = integrator->getDxDp();
+    RCP<const Thyra::VectorBase<double> > x_exact =
+      model->getExactSolution(time).get_x();
+    RCP<Thyra::MultiVectorBase<double> > DxDp_exact =
+      Thyra::createMembers(model->get_x_space(), num_param);
+    for (int i=0; i<num_param; ++i)
+      Thyra::assign(DxDp_exact->col(i).ptr(),
+                    *(model->getExactSensSolution(i, time).get_x()));
+
+    // Plot sample solution and exact solution
+    if (comm->getRank() == 0 && n == nTimeStepSizes-1) {
+      typedef Thyra::DefaultMultiVectorProductVector<double> DMVPV;
+
```

Comment:
This broke pull request testing for a later pull request, see [here](/ascldap/users/daibane/src/alexa/src/LowRmPotentialSolve.hpp)